src/components/MainComposer/components/ComposerEditor/ComposerEditor.tsx

```diff
@@ -1,56 +1,79 @@
 import { ChangeEvent, ReactNode, useCallback, useState } from 'react';
 
+import omit from 'lodash.omit';
+import { nanoid } from 'nanoid';
+
 import {
   PostMode,
   TextValidator,
 } from '~services/api/social-media/social-media.types.ts';
 import { useAccountStore } from '~stores/useAccountStore/useAccountStore';
 import { useSocialMediaStore } from '~stores/useSocialMediaStore/useSocialMediaStore';
 
-import { TextValidators } from './utils/textValidator/textValidators';
+import { textValidator } from './utils/textValidator/textValidator';
 
 import CharacterLimit from '~components/CharacterLimitMainText/CharacterLimit';
 
 import scss from './ComposerEditor.module.scss';
 
+import { Error } from '../MainComposerBase/MainComposerBase.type';
 import {
   ComposerEditorProps,
-  ErrorMapText,
-  ErrorText,
   HigherLimitSocial,
+  TEXT_ERRORS,
+  TextErrorMap,
 } from './ComposerEditor.types';
 
 function ComposerEditor(props: ComposerEditorProps): ReactNode {
   const { socialMedias } = useSocialMediaStore();
   const { updateMainContent } = useAccountStore();
   const [inputValue, setInputValue] = useState('');
-  const [errorMap, setErrorMap] = useState<ErrorMapText>({});
-
-  const validatorText = (text: string): ErrorText[] => {
-    const textValidators = new TextValidators(text);
-    const validators = props.postMode?.validators as TextValidator;
-    const errors: ErrorText[] = [];
-
-    if (
-      props.postMode &&
-      textValidators.textLength(validators.text.maxLength)
-    ) {
-      errors.push({
-        accountId: props.accountId,
-        message: 'text exceeded the limit',
-        postModeId: props.postMode.id,
-      });
+  const [errors, setErrors] = useState<TextErrorMap>();
+
+  const addErrors = (textErrors: TEXT_ERRORS, errorForAdd: Error): void => {
+    const errorId = nanoid();
+    const newError = { ...errors };
+    const error = newError[textErrors];
+
+    if (!error) {
+      setErrors({ ...errors, [textErrors]: errorId });
+      props.addError?.(errorId, errorForAdd);
     }
```

Comment:
Não precisamos desse spread antes pq usamos o `error` apenas para uma verificação
```suggestion
    const errorId = nanoid();
    const error = errors[textErrors];

    if (!error) {
      setErrors({ ...errors, [textErrors]: errorId });
      props.addError?.(errorId, errorForAdd);
    }
```